static/app/components/searchSyntax/renderer.tsx

```diff
@@ -130,6 +127,19 @@ function FilterToken({
     );
   }, [reduceMotion, showInvalid]);
 
+  return {filterElementRef, showInvalid, showWarning, showTooltip, isActive};
+}
+
+function FilterToken({
+  filter,
+  cursor,
+}: {
+  cursor: number;
+  filter: TokenResult<Token.FILTER>;
+}) {
+  const {showInvalid, showTooltip, showWarning, filterElementRef, isActive} =
+    useTokenValidation(cursor, filter);
```

Comment:
Nice extraction 👍

Reply:
Thanks for doing this I remember this got duplicated and I had wanted to clean it up